aineko/extras/connectors/wss.py

```diff
@@ -0,0 +1,160 @@
+# Copyright 2023 Aineko Authors
+# SPDX-License-Identifier: Apache-2.0
+"""Extra module for connecting to a WebSocket."""
+
+import json
+import time
+from typing import Any, Dict, List, Optional
+
+import websocket
+from pydantic import BaseModel, field_validator
+
+from aineko import AbstractNode
+from aineko.utils.secrets import inject_secrets
+
+
+class ParamsWSS(BaseModel):
+    """Connector params for WebSocket model."""
+
+    max_retries: int = 30
+    retry_sleep: float = 5
+    url: str
+    header: Optional[Dict[str, str]] = None
+    init_messages: List[Any] = []
+    metadata: Optional[Dict[str, Any]] = None
+
+    @field_validator("url")
+    def supported_url(cls, u: str) -> str:  # pylint: disable=no-self-argument
+        """Validates that the url is a valid WebSocket URL."""
+        if not u.startswith("wss://") or u.startswith("ws://"):
+            raise ValueError(
+                "Invalid url provided to WebSocket params. "
+                'Expected url to start with "wss://" or "ws://". '
+                f"Provided url was: {str(u)}"
+            )
+        return u
+
+
+class WSS(AbstractNode):
+    """Connector for a websocket."""
+
+    retry_count = 0
+
+    def _pre_loop_hook(self, params: dict | None = None) -> None:
```

Comment:
```suggestion
    def _pre_loop_hook(self, params: Optional[dict] = None) -> None:
```

I believe this is the syntax we should use based on our style conventions.

Reply:
I did this because I was failing mypy checks locally, but i think i had some stale settings. i fixed this here: https://github.com/aineko-dev/aineko/pull/121/commits/a4db5b4f648ceec3478a51d71eaff97b55383ac2

Reply:
No the original code was correct. We're now using python 3.10 type hints https://github.com/aineko-dev/aineko/pull/119

Reply:
reverted here: https://github.com/aineko-dev/aineko/pull/121/commits/86629c74f4be340b4cfa42a0df907f348ea628b6

Thanks @Barabazs